rimage/image_file.go

```diff
@@ -309,8 +329,17 @@ func EncodeImage(ctx context.Context, img image.Image, mimeType string) ([]byte,
 			return nil, err
 		}
 	case ut.MimeTypeJPEG:
-		if err := jpeg.Encode(&buf, img, nil); err != nil {
-			return nil, err
+		switch v := img.(type) {
+		case *Image:
+			imgRGBA := image.NewRGBA(bounds)
+			ConvertToRGBA(imgRGBA, v)
+			if err := libjpeg.Encode(&buf, imgRGBA, &libjpeg.EncoderOptions{Quality: 75, DCTMethod: libjpeg.DCTIFast}); err != nil {
+				return nil, err
+			}
+		default:
+			if err := libjpeg.Encode(&buf, img, &libjpeg.EncoderOptions{Quality: 75, DCTMethod: libjpeg.DCTIFast}); err != nil {
```

Comment:
I'm assuming you tried and can't simply use a *rimage.Image directly even though it fulfills the image.Image interface.

Reply:
Exactly. The functions that are encoding are different from one color model to another.